### PR TITLE
[MAINTENANCE] Splitter cleanup and enhancements

### PR DESCRIPTION
**Pull request by anthonyburdi**
Changes proposed in this pull request:
- [Use 'multi' method in to_sql for redshift](https://github.com/great-expectations/great_expectations/commit/43d6afe48bf0cb2063932a826a9e46397a3aab84)
- [Move sqlalchemy_data_splitter.py into execution_engine/split_and_samp…](https://github.com/great-expectations/great_expectations/commit/c0e7a477cf5fb101f6386a77b22cb5183c2730a7) 
- [Remove underscore prefix from examples](https://github.com/great-expectations/great_expectations/commit/1dbaff75d92b7408175205276f212d08d4c6b344)
- Move logic and previously existing splitter methods out of ConfiguredAssetSqlDataConnector and into SqlDataSplitter.
- Ensure that previously existing methods work with and without underscore via updating tests.
- Simplify interface to get data for batch identifiers via single method `SqlAlchemyDataSplitter. get_data_for_batch_identifiers()` called via `SqlAlchemyExecutionEngine. get_data_for_batch_identifiers()` in the data connector.

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: tests/integration/db/test_sql_data_splitting.py [hunk 1]
```diff
@@ -49,9 +49,18 @@ def _get_connection_string_and_dialect() -> Tuple[str, str]:
...
-    connection_string: str, table_name: str = TAXI_DATA_TABLE_NAME
+    connection_string: str, dialect: str, table_name: str = TAXI_DATA_TABLE_NAME
 ) -> LoadedTable:
 
+    dialects_supporting_multiple_values_in_single_insert_clause: List[str] = [
+        "redshift"
+    ]
+    to_sql_method: str = (
+        "multi"
+        if dialect in dialects_supporting_multiple_values_in_single_insert_clause
```
Comment (Shinnnyshinshin) [inline tests/integration/db/test_sql_data_splitting.py:60]: ❤️

Comment (Shinnnyshinshin) [reply tests/integration/db/test_sql_data_splitting.py:60]: I thought this was a really elegant solution to the issue. Thank you @anthonyburdi

**Review by alexsherstinsky — APPROVED** [review]
LGTM

**Comment by Shinnnyshinshin** [comment]
I see one failing test for `docs-integration`. Maybe a attribute name needs to be adjusted from private to public? 

`great_expectations.exceptions.exceptions.DatasourceInitializationError: Cannot initialize datasource taxi_datasource, error: 'InferredAssetSqlDataConnector' object has no attribute 'split_on_converted_datetime'`

**Review by Shinnnyshinshin — CHANGES_REQUESTED** [review]
Very close. One failing integration test I believe. Otherwise looks great :)

**Review by cdkini — APPROVED** [review]
LGTM!

File: great_expectations/datasource/data_connector/configured_asset_sql_data_connector.py [hunk 2]
```diff
@@ -230,7 +214,7 @@ def _get_batch_identifiers_list_from_data_asset_config(
             else:
                 batch_identifiers_list: List[
                     dict
-                ] = self._get_batch_identifiers_from_data_connector_method(
+                ] = self.execution_engine.get_data_for_batch_identifiers_for_other_splitters(
                     table_name, splitter_method_name, splitter_kwargs
```
Comment (Shinnnyshinshin) [inline great_expectations/datasource/data_connector/configured_asset_sql_data_connector.py]: ❤️

File: tests/datasource/data_connector/test_sql_data_connector.py [hunk 3]
```diff
@@ -74,20 +74,22 @@ def test_basic_self_check(test_cases_for_sql_data_connector_sqlite_execution_eng
...
     config = yaml.load(
-        """
+        f"""
     name: my_sql_data_connector
     datasource_name: FAKE_Datasource_NAME
 
     assets:
         table_partitioned_by_date_column__A:
-            splitter_method: _split_on_column_value
+            splitter_method: {splitter_method_name_prefix}split_on_column_value
```
Comment (Shinnnyshinshin) [inline tests/datasource/data_connector/test_sql_data_connector.py:92]: super interesting and elegant way to deal with this @anthonyburdi . Love what you're doing with introducing parameterized tests.

**Review by Shinnnyshinshin — APPROVED** [review]
LGTM @anthonyburdi thank you !

**Pull request merged**